library/sql_upgrade_fx.php

```diff
@@ -597,14 +597,14 @@ function convertLayoutProperties()
 *
 * @param  string  $filename  Sql upgrade/patch filename
 */
-function upgradeFromSqlFile($filename)
+function upgradeFromSqlFile($filename, $path = '')
 {
     global $webserver_root;
 
     flush();
     echo "<font color='green'>Processing $filename ...</font><br />\n";
 
-    $fullname = "$webserver_root/sql/$filename";
+    $fullname = (is_dir($path) ? $path : $webserver_root)."/sql/$filename";
```

Comment:
check if !empty and then is_dir:
```
((!empty($path) && is_dir($path)) ? $path : $webserver_root)
```
And place a quick comment above this line.

Reply:
I will fix it